### PR TITLE
[APIView] Disable build step `Create API Review` for private repos

### PR DESCRIPTION
**Pull request by maririos**
Fixes https://github.com/Azure/azure-sdk-tools/issues/3427

### CODE REVIEW

**Review by praveenkuttappan — APPROVED** [review]
Looks good

**Review by weshaggard — APPROVED** [review]
I think this will work for now. However we might need another solution later if we find that the "-pr" naming convention isn't a strong enough condition.

**Comment by maririos** [comment]
@weshaggard Agree. I also looked into all steps to see which ones had a similar behavior and so far only 2 (including this one) have it, so decided it was ok for now. If we start having more, then maintaining the behavior on individual steps might not be ideal.

File: eng/common/pipelines/templates/steps/create-apireview.yml [hunk 1]
```diff
@@ -24,4 +24,4 @@ steps:
         pwsh: true
         workingDirectory: $(Pipeline.Workspace)
       displayName: Create API Review for ${{ artifact.name}}
-      condition: and(succeededOrFailed(), ne(variables['Skip.CreateApiReview'], 'true') , ne(variables['Build.Reason'],'PullRequest'), eq(variables['System.TeamProject'], 'internal'))
+      condition: and(succeededOrFailed(), ne(variables['Skip.CreateApiReview'], 'true') , ne(variables['Build.Reason'],'PullRequest'), eq(variables['System.TeamProject'], 'internal'), not(endsWith(variables['Build.Repository.Name'], '-pr')))
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/create-apireview.yml:27]: Might be worth splitting this condition across multiple lines to help with readability. Something like https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/pipelines/templates/jobs/ci.tests.yml#L91

Comment (maririos) [reply eng/common/pipelines/templates/steps/create-apireview.yml]: Thank you. Updated

**Comment by weshaggard** [comment]
@maririos this will be your first eng/common PR. There is a link with instructions on how to process it but let me know if you need any help.

**Pull request merged**